Unable to override home directory

When building/testing in a containerized environment (such as Nix), the default home directory (`/var/empty`) is inaccessible. This prevents the unit tests from functioning:

```
dcrd> ++ go test -p 12 ./. -args --appdata /private/tmp/nix-build-dcrd-1.8.0.drv-1/tmp.G5ZHoVaIR3
dcrd> 2023-06-28 20:49:03.438 [INF] RPCS: Generating TLS certificates...
dcrd> 2023-06-28 20:49:03.457 [INF] RPCS: Done generating TLS certificates
dcrd> 2023-06-28 20:49:03.457 [INF] RPCS: Generating TLS certificates...
dcrd> 2023-06-28 20:49:03.464 [INF] RPCS: Done generating TLS certificates
dcrd> Error creating a default config file: mkdir /var/empty: file exists
dcrd> --- FAIL: TestLoadConfig (0.00s)
dcrd>     config_test.go:30: Failed to load dcrd config: loadConfig: failed to create home directory: mkdir /var/empty: file exists
dcrd> Error creating a default config file: mkdir /var/empty: file exists
dcrd> --- FAIL: TestDefaultAltDNSNames (0.00s)
dcrd>     config_test.go:41: Failed to load dcrd config: loadConfig: failed to create home directory: mkdir /var/empty: file exists
dcrd> Error creating a default config file: mkdir /var/empty: file exists
dcrd> --- FAIL: TestAltDNSNamesWithEnv (0.00s)
dcrd>     config_test.go:56: Failed to load dcrd config: loadConfig: failed to create home directory: mkdir /var/empty: file exists
dcrd> Error creating a default config file: mkdir /var/empty: file exists
dcrd> --- FAIL: TestAltDNSNamesWithArg (0.00s)
dcrd>     config_test.go:74: Failed to load dcrd config: loadConfig: failed to create home directory: mkdir /var/empty: file exists
dcrd> FAIL
dcrd> FAIL      github.com/decred/dcrd  0.190s
dcrd> FAIL
```

The logic in [`appDataDir`](https://github.com/decred/dcrd/blob/cc33b719bca49c8e9e8a048b8c6bef9418dd9e9d/dcrutil/appdata.go#L22) will only consult the `HOME` environment variable if the lookup from `/etc/passwd` fails. This prevents me from overriding the value at runtime. If the `HOME` environment variable was consulted first, running the unit tests in a sandbox would be much easier.

--- a/config_test.go
+++ b/config_test.go
@@ -1,15 +1,26 @@
-// Copyright (c) 2018 The Decred developers
+// Copyright (c) 2018-2023 The Decred developers
 // Use of this source code is governed by an ISC
 // license that can be found in the LICENSE file.
 
 package main
 
 import (
+	"flag"
 	"os"
 	"path/filepath"
 	"strings"
 	"testing"
 )
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	// Trim -test.* flags from the command line arguments list to allow
+	// go-flags tests to succeed.
+	os.Args = append([]string{os.Args[0]}, flag.Args()...)
+
+	os.Exit(m.Run())
+}
 
 // In order to test command line arguments and environment variables, append
 // the flags to the os.Args variable like so:
@@ -79,10 +90,4 @@ func TestAltDNSNamesWithArg(t *testing.T) {
 			hostnames)
 	}
 	os.Args = old
-}
-
-// init parses the -test.* flags from the command line arguments list and then
-// removes them to allow go-flags tests to succeed.
-func init() {
-	os.Args = os.Args[:1]
 }

